nonthermal.cc

```diff
@@ -467,7 +467,64 @@ void init(const int my_rank, const int ndo_nonempty) {
     fflush(nonthermalfile);
   }
 
+  int LIST_LENGTH = MAX_NT_EXCITATIONS_STORED;
+#ifdef MPI_ON
+
+  MPI_Win nt_solution_window;
+
+  // We want to keep the nt_solution_struct in local but the nt_excitation_struct in shared memory
+  int rank, shm_rank;
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+  // Get the rank of the shared memory communicator
+  MPI_Comm shmcomm;
+  MPI_Comm_split_type(MPI_COMM_WORLD, MPI_COMM_TYPE_SHARED, 0, MPI_INFO_NULL, &shmcomm);
+
+  // Rank within the shared memory communicator
+  MPI_Comm_rank(shmcomm, &shm_rank);
```

Comment:
These lines are redundant since we already have the rank in globals::rank_global and the shared memory communicator is globals::mpi_comm_node. It's especially important to use globals::mpi_comm_node since on the CI jobs, these "nodes" are further split into subsets of the shared memory system in order to test internode and intranode communication.

Reply:
Changed allocation of memory from using shmcomm to globals::rank_in_node